CarpetX/src/schedule.cxx

```diff
@@ -1516,26 +1516,18 @@ void CycleTimelevels(cGH *restrict const cctkGH) {
         // All time levels (except the current) must be valid everywhere for
         // checkpointed groups
         bool presync_only = CCTK_EQUALS(presync_mode, "presync-only");
-        if (presync_only && groupdata.do_checkpoint) {
-          std::set<int> sync_set;
-          for (int tl = (ntls == 1 ? 0 : 1); tl < ntls; ++tl) {
-            for (int vi = 0; vi < groupdata.numvars; ++vi) {
-              sync_set.insert(gi);
-            }
-          }
-          if (!sync_set.empty()) {
-            std::vector<int> sync_vec(sync_set.begin(), sync_set.end());
-            SyncGroupsByDirI(cctkGH, sync_vec.size(), sync_vec.data(), nullptr);
-          }
-        }
-        // check that what we have is correct
         if (groupdata.do_checkpoint) {
-          std::set<int> sync_set;
           for (int tl = (ntls == 1 ? 0 : 1); tl < ntls; ++tl) {
             for (int vi = 0; vi < groupdata.numvars; ++vi) {
-              error_if_invalid(groupdata, vi, tl, make_valid_all(), []() {
-                return "CycleTimelevels for the state vector";
-              });
+              if(presync_only) {
+                std::vector<int> sync_vec;
+                sync_vec.push_back(gi);
+                SyncGroupsByDirI(cctkGH, sync_vec.size(), sync_vec.data(), nullptr);
```

Comment:
that only ever has a single group in `sync_vec` and then will sync that group `groupdata.numvars` times, won't it?

Much better would seem to collect all groups that need syncs in a `for (int gi = 0; gi < num_groups; ++gi) {` first, then sync them all.